### ISSUE TITLE
"Array of Choices" text input field not appear when wanting to write an expression for multi-choice prompt

### ISSUE THREAD

**Issue by alfenne** [microsoft/BotFramework-Composer#6651]
## Describe the bug

After creating a new project with the "New Creation Experience" box unchecked, I am unable to access the "Array of choices" text field after selecting "Write an Expression" from the drop down option when trying to create a multi-choice prompt.

## Version

1.4-nightly.229963.f72f063

## Browser

- [ ] Electron distribution

## OS

- [ ] Windows

## To Reproduce

1. In the "Composer Settings" tab ensure that the "New Creation Experience" toggle is unchecked.
2. Create a bot from scratch
3. Once the bot is scaffolded, create a multi-choice prompt anywhere in the dialog flow
4. In the side bar that appears, navigate to the "User Input" tab
5. On the "Array of Choices" field, select =Write an Expression from the dropdown list
6. Finding that the text field for writing an expression is missing

## Expected behavior

When selecting =Write an Expression from the "Array of Choices" dropdown, a text field to write an expression should appear.

## Screenshots

<img width="275" alt="Screenshot 2021-04-01 142121" src="https://user-images.githubusercontent.com/64445981/113337595-fc272a00-92f5-11eb-9e1e-461fabac5ba5.png">

**Comment by alfenne** [comment]
@garypretty

**Comment by msomanathan** [comment]
@cwhitten Could you please check if Choice prompt has an issue in composer?

**Comment by cwhitten** [comment]
@tdurnford will be looking at this

**Comment by tdurnford** [comment]
@alfenne @cwhitten I’m able to reproduce this issue. Still investigating why the field isn’t resolving correctly when the value is undefined. I think it has something to do with intellisense, but I’m not sure.

**Comment by garypretty** [comment]
@tdurnford not sure if this is related, but if I try and set a dialogs options to 'object', I am getting a key / value list instead of the JSON input. 

![image](https://user-images.githubusercontent.com/3900649/113871018-5cace000-97aa-11eb-9c31-61eb2e25530b.png)

**Comment by tdurnford** [comment]
@garypretty That is not related to this issue, but to comment on that issue, we are rendering the OpenObjectField instead of the JsonField because that is the default field for the dialog options property. Also, I tried using the JsonField instead; however, when I used the JsonField, the other forms would not update the dialog file correctly. I was unable to figure out what was causing that issue, but if you want to cut another ticket for it, I’d be happy to take a look. I agree the JsonField would be a better editing experience than the OpenObjectField in that scenario.
